result/arp.yaml

```diff
@@ -0,0 +1,73 @@
+openapi: 3.1.0
+
+info:
+  title: ARP table models
+  version: ^0.0.0
+
+components:
+  schemas:
+    Arp.Entries.Request:
+      description: >-
+        The request to retrieve ARP entries for IPv4 interface(s).
+      type: object
+      properties:
+        ipv4_names:
+          description: >-
+            The names of IPv4 interfaces for which ARP entries will be retrieved.
+            If no interface names are specified then the result set will contain all ARP entries for all available IPv4 interfaces.
+          type: array
+          items:
+            type: string
+          x-constraint:
+            - "/components/schemas/Device.Ipv4/properties/name"
+        column_names:
+          description: >-
+            Name of the columns (as list), that will present in the returned result set.
+            If the column names is empty then all columns will be returned.
+            The column ipv4_name (the name of the IPv4 interface) will always be present in the returned result set.
+          type: array
+          items:
+            type: string
+            enum:
+              - ipv4_name
+              - neighbor_ip
+              - link_layer_address
+
+    Arp.Entries.Response:
+      description: >-
+        A container for ARP entries.
+      type: object
+      properties:
```

Comment:
Why don't we just add `resolved` as a column name where its value (true / false) for any given row should indicate the same info ?

IMHO, additional nesting of resolved vs unresolved is redundant here.

Reply:
> Why don't we just add `resolved` as a column name where its value (true / false) for any given row should indicate the same info ?
> 
> IMHO, additional nesting of resolved vs unresolved is redundant here.

What is the use-case for providing a list of unresolved entries in the first place?  What is that list expected to contain?  Just interface_name?

Reply:
I can't imagine a scenario where an end user (in a script) is going to go thru 1000 entries of an ARP table to see what resolved and what didn't.  And the way to determine if an IP is resolved or not, is just to go thru the resolved list to see if it is present there or not.  I don't know of any tool (or OS) that gives a list of unresolved IPs. 

I still think we are missing the use-case(s) around this feature.  

I don't think that ARP will be a test-case by itself.  Usually, there will be other protocols that will be running on top of IP and those will fail.  Typically, at that time, ARP is used as a diagnostic tool and that works well from a GUI app.  In case of an API driven environment, I'm not sure what value the ARP table would serve.

That said, I think a more important piece that _might_ be missing is the ability for the user to trigger ARP for a specific destination IP.  

Let's get the use-cases and all these questions will be automatically answered.